aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -123,6 +125,18 @@ void startSubscriptions() {
         .onErrorResumeNext(next -> {
             next.onComplete();
         })
+        .doOnDispose(() -> {
+            // When the observable is disposed, we need to call cancel() on the subscription
+            // so it can properly dispose of resources if necessary. For the AWS API plugin,
+            // this means means closing the underlying network connection.
+            synchronized (cancelable) {
+                Cancelable subscriptionOperation = cancelable.get();
+                if (subscriptionOperation != null) {
+                    LOG.info("Terminating subscription operation.");
+                    subscriptionOperation.cancel();
+                }
+            }
+        })
```

Comment:
L108 was _supposed_ to do this.

```
            emitter.setCancellable(cancelable::get);
```

However, I think I found elsewhere, that `setCancellable(...)` doesn't do what I thought it was supposed to do. There is also `emitter.setDispoable(Disposable...)`, though. I have a factory method elsewhere in the code base that builds a `Disposable` from our own `Cancelable` type. There's a note about "pulling it out into a utility class". Maybe its time.....